packages/core/src/compile.ts

```diff
@@ -149,9 +149,10 @@ function serialiseTranslationRuntime(
 
     if (Object.keys(params).length > 0) {
       const formatGeneric = hasTags ? '<T = string>' : '';
-      const formatReturn = hasTags
-        ? 'string | T | Array<string | T>'
-        : 'string';
+      const formatReturn =
+        hasTags && imports.has('FormatXMLElementFn')
```

Comment:
Should `'FormatXMLElementFn'` be added to `imports` if `hasTags` is true? Or is that already accounted for somewhere else? Maybe the addition and check should be done at the same time?

Reply:
No, it was already added to `imports` in a previous step.